common/src/main/java/com/wynntils/mc/extension/ChatComponentExtension.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.extension;
+
+import net.minecraft.network.chat.Component;
+
+public interface ChatComponentExtension {
+    void deleteMessage(Component component);
```

Comment:
Ohhh... I wonder if this framework can help us restore BACKGROUND chat messages (grayscale versions sent during a chat) with FOREGROUND versions (normally colored that are "restored" when the chat is ended). That is currently actually a regression compared to vanilla Wynncraft, that we do not restore colors afterwards.

Just thinking out loud.

Reply:
We could, rather easily.. I was just not sure if I should take that step here, or check if these changes are stable by themselves.

I think I won't do this, until I fix up client messages with chat tabs, since currently we don't really have great support for chat tabs even in our own code. (client side messages are sent only to the currently focused tab)

Reply:
Yeah, not suggesting you do it here. Keep it as a separate change.